scenario/outputs.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""
+Output types.
+Objects that wrap trigger outcomes which don't quite fit in State, as they can't be used as
+input for the next trigger.
+"""
+
+from contextvars import ContextVar
+
+# If you are using pytest, the scenario.pytest_plugin.action_output fixture should take care
+# of managing this variable and set/reset it once per test.
+# If you are not using pytest, then you'll need to .set() this var to a new ActionOutput
+# instance before each Context.run() and reset it when you're done.
+ACTION_OUTPUT = ContextVar("ACTION_OUTPUT")
```

Comment:
I don't understand why this global is necessary? (or is it a thread-local; I'm not familiar with contextvars). Why isn't an instance of `ActionOutput` enough?

Reply:
It's basically a global variable, but yes, thread local (not that I'm using that feature of it here (yet)).
Reason why I like it is that it has a nice api for `get()`ting and `set()`ting it so that attempting to get it before it's been set raises an exception. So I don't need to initialize it with a default, basically, or have some code to check whether it's been set before I attempt to read it.